llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1469,6 +1465,47 @@ void VPReplicateRecipe::print(raw_ostream &O, const Twine &Indent,
 }
 #endif
 
+Value *VPScalarCastRecipe ::generate(VPTransformState &State, unsigned Part) {
+  assert(vputils::onlyFirstLaneUsed(this) &&
+         "Codegen only implemented for first lane only.");
```

Comment:
The implementation here is intentionally kept generic for any cast where only the first lane is used, so it can be used directly in https://github.com/llvm/llvm-project/pull/76172